server-common/src/main/java/org/apache/kafka/server/config/QuotaConfigs.java

```diff
@@ -0,0 +1,154 @@
+            "This property can be only set dynamically. It is suggested that the limit be kept above 1MB/s for accurate behaviour.";
+    public static final long QUOTA_BYTES_PER_SECOND_DEFAULT = Long.MAX_VALUE;
+
+    public static final String PRODUCER_BYTE_RATE_OVERRIDE_CONFIG = "producer_byte_rate";
+    public static final String CONSUMER_BYTE_RATE_OVERRIDE_CONFIG = "consumer_byte_rate";
+    public static final String REQUEST_PERCENTAGE_OVERRIDE_CONFIG = "request_percentage";
+    public static final String CONTROLLER_MUTATION_RATE_OVERRIDE_CONFIG = "controller_mutation_rate";
+    public static final String IP_CONNECTION_RATE_OVERRIDE_CONFIG = "connection_creation_rate";
+    public static final String PRODUCER_BYTE_RATE_DOC = "A rate representing the upper bound (bytes/sec) for producer traffic.";
+    public static final String CONSUMER_BYTE_RATE_DOC = "A rate representing the upper bound (bytes/sec) for consumer traffic.";
+    public static final String REQUEST_PERCENTAGE_DOC = "A percentage representing the upper bound of time spent for processing requests.";
+    public static final String CONTROLLER_MUTATION_RATE_DOC = "The rate at which mutations are accepted for the create " +
+            "topics request, the create partitions request and the delete topics request. The rate is accumulated by " +
+            "the number of partitions created or deleted.";
+    public static final String IP_CONNECTION_RATE_DOC = "An int representing the upper bound of connections accepted " +
+            "for the specified IP.";
+
+    public static final int IP_CONNECTION_RATE_DEFAULT = Integer.MAX_VALUE;
+
+    private final static Set<String> USER_AND_CLIENT_QUOTA_NAMES = new HashSet<>(Arrays.asList(
+            PRODUCER_BYTE_RATE_OVERRIDE_CONFIG,
+            CONSUMER_BYTE_RATE_OVERRIDE_CONFIG,
+            REQUEST_PERCENTAGE_OVERRIDE_CONFIG,
+            CONTROLLER_MUTATION_RATE_OVERRIDE_CONFIG
+    ));
+
+    private static void buildUserClientQuotaConfigDef(ConfigDef configDef) {
+        configDef.define(PRODUCER_BYTE_RATE_OVERRIDE_CONFIG, ConfigDef.Type.LONG, Long.MAX_VALUE,
+                ConfigDef.Importance.MEDIUM, PRODUCER_BYTE_RATE_DOC);
+
+        configDef.define(CONSUMER_BYTE_RATE_OVERRIDE_CONFIG, ConfigDef.Type.LONG, Long.MAX_VALUE,
+                ConfigDef.Importance.MEDIUM, CONSUMER_BYTE_RATE_DOC);
+
+        configDef.define(REQUEST_PERCENTAGE_OVERRIDE_CONFIG, ConfigDef.Type.DOUBLE,
+                Integer.valueOf(Integer.MAX_VALUE).doubleValue(),
+                ConfigDef.Importance.MEDIUM, REQUEST_PERCENTAGE_DOC);
+
+        configDef.define(CONTROLLER_MUTATION_RATE_OVERRIDE_CONFIG, ConfigDef.Type.DOUBLE,
+                Integer.valueOf(Integer.MAX_VALUE).doubleValue(),
+                ConfigDef.Importance.MEDIUM, CONTROLLER_MUTATION_RATE_DOC);
+    }
+
+    public static boolean isClientOrUserConfig(String name) {
```

Comment:
This method is used by `isClientOrUserQuotaConfig` only, so they can be merged in one method

Reply:
Good call just updated this